DOCSP-41958 - Connection Targets (#107)
mongodb/docs-php-library: source/connect/connection-targets.txt

.. _php-connection-targets:

==========================
Choose a Connection Target
==========================

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: connection string, URI, server, settings, client, stable api

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

Overview
--------

In this guide, you can learn how to use a connection string and ``MongoDB\Client`` object
to connect to different types of MongoDB deployments.

.. _php-connection-atlas:

Atlas
-----

To connect to a MongoDB deployment on Atlas, include the following elements
in your connection string:

- URI of your Atlas cluster

> - MongoDB username
> - MongoDB password

Reviewer: S: just to clarify that this isn't your Atlas login; I think just saying "MongoDB" could be confusing (related to https://jira.mongodb.org/browse/DOCSP-42303)

```suggestion
- Database username
- Database user's password
```

Then, pass your connection string to the ``MongoDB\Client`` constructor.

When you connect to Atlas, we recommend using the {+stable-api+} client option to avoid
breaking changes when Atlas upgrades to a new version of {+mdb-server+}.
To learn more about the {+stable-api+} feature, see the :ref:`{+stable-api+} page
<php-stable-api>`.

The following code shows how to use the {+driver-short+} to connect to an Atlas cluster.
The code also uses the ``serverApi`` option to specify a {+stable-api+} version.

.. literalinclude:: /includes/connect/atlas.php
   :copyable: true
   :language: php

.. tip::

>    Follow the :atlas:`Atlas driver connection guide </driver-connection>`

Reviewer: S: you can link to the Connection String step of the quick start here instead
```suggestion
   Follow the :ref:`php-connection-string` step of the Quick Start
```

Author: good idea!

   to retrieve your connection string.

.. _php-connection-local:

Local Deployments
-----------------

To connect to a local MongoDB deployment, use ``localhost`` as the hostname. By
default, the ``mongod`` process runs on port 27017, though you can customize this for
your deployment.

The following code shows how to use the {+driver-short+} to connect to a local MongoDB
deployment:

.. literalinclude:: /includes/connect/client.php
   :language: php
   :copyable: true

.. _php-connection-replica-set:

Replica Sets
------------

To connect to a replica set, specify the hostnames (or IP addresses) and 
port numbers of the replica set members in your connection string.

If you aren't able to provide a full list of hosts in the replica set, you can 
specify one or more of the hosts in the replica set and instruct the {+driver-short+} to 
perform automatic discovery to find the others. To instruct the driver to perform

> automatic discovery, perform one of the following actions:

Reviewer: S: since this sentence repeats "perform":

```suggestion
automatic discovery, choose one of the following actions:
```

.. _php-connection-targets:

==========================
Choose a Connection Target
==========================

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: connection string, URI, server, settings, client, stable api

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

Overview
--------

In this guide, you can learn how to use a connection string and ``MongoDB\Client`` object
to connect to different types of MongoDB deployments.

.. _php-connection-atlas:

Atlas
-----

To connect to a MongoDB deployment on Atlas, include the following elements
in your connection string:

- URI of your Atlas cluster
- MongoDB username
- MongoDB password

Then, pass your connection string to the ``MongoDB\Client`` constructor.

When you connect to Atlas, we recommend using the {+stable-api+} client option to avoid
breaking changes when Atlas upgrades to a new version of {+mdb-server+}.
To learn more about the {+stable-api+} feature, see the :ref:`{+stable-api+} page

> <php-stable-api>`.

Author: broken link for now

The following code shows how to use the {+driver-short+} to connect to an Atlas cluster.

The code also uses the ``server_api_opts`` option to specify a {+stable-api+} version.

.. literalinclude:: /includes/connect/atlas.php
   :copyable: true
   :language: php

.. tip::

   Follow the :atlas:`Atlas driver connection guide </driver-connection>`
   to retrieve your connection string.

.. _php-connection-local:

Local Deployments
-----------------

To connect to a local MongoDB deployment, use ``localhost`` as the hostname. By
default, the ``mongod`` process runs on port 27017, though you can customize this for
your deployment.

The following code shows how to use the {+driver-short+} to connect to a local MongoDB
deployment:

> .. literalinclude:: /includes/connect/client.php

Author: broken include for now
